gcc/rust/resolve/rust-ast-resolve-toplevel.h

```diff
@@ -39,6 +41,29 @@ class ResolveTopLevel : public ResolverBase
     item->accept_vis (resolver);
   };
 
+  void visit (AST::ModuleBodied& module) override {
+    auto path = prefix.append (CanonicalPath(module.get_name()));
+    resolver->get_name_scope ().insert (
+      path, module.get_node_id (), module.get_locus (), false,
```

Comment:
I think the name resolution stuff could be extracted into a single commit a test case like:

```rust
mod foo {
  struct A;
}

fn main() {}
```

Might work since the path expression stuff will not be invoked.

Reply:
Done, added in a dedicated test and checked it works with only the changes for name resolution applied.